core/src/test/java/io/grpc/util/GracefulSwitchLoadBalancerTest.java

```diff
@@ -329,19 +336,51 @@ public void switchWhileOldPolicyIsReadyThenOldPolicyGoesToNotReady() {
 
     LoadBalancer lb1 = balancers.get(lbPolicies[1]);
     Helper helper1 = helpers.get(lb1);
+    SubchannelPicker picker1 = mock(SubchannelPicker.class);
+    helper1.updateBalancingState(CONNECTING, picker1);
+    verify(mockHelper, never()).updateBalancingState(CONNECTING, picker1);
+
     picker = mock(SubchannelPicker.class);
-    helper1.updateBalancingState(CONNECTING, picker);
+    helper0.updateBalancingState(CONNECTING, picker);
+    verify(lb0).shutdown();
     verify(mockHelper, never()).updateBalancingState(CONNECTING, picker);
+    verify(mockHelper).updateBalancingState(CONNECTING, picker1);
+
+    picker1 = mock(SubchannelPicker.class);
+    helper1.updateBalancingState(READY, picker1);
+    verify(mockHelper).updateBalancingState(READY, picker1);
+
+    verifyNoMoreInteractions(lb0, lb1);
+  }
+
+  @Test
+  public void switchWhileOldPolicyGoesFromReadyToNotReadyWhileNewPolicyStillIdle() {
+    gracefulSwitchLb.switchTo(lbProviders.get(lbPolicies[0]));
+    LoadBalancer lb0 = balancers.get(lbPolicies[0]);
+    InOrder inOrder = inOrder(lb0, mockHelper);
+    Helper helper0 = helpers.get(lb0);
+    SubchannelPicker picker = mock(SubchannelPicker.class);
+    helper0.updateBalancingState(READY, picker);
+
+    gracefulSwitchLb.switchTo(lbProviders.get(lbPolicies[1]));
+    inOrder.verify(lb0, never()).shutdown();
+
+    LoadBalancer lb1 = balancers.get(lbPolicies[1]);
+    Helper helper1 = helpers.get(lb1);
 
     picker = mock(SubchannelPicker.class);
     helper0.updateBalancingState(CONNECTING, picker);
-    verify(lb0).shutdown();
+
+    inOrder.verify(lb0).shutdown();
+    inOrder.verify(mockHelper, never()).updateBalancingState(CONNECTING, picker);
```

Comment:
Does `inOrder.verify(, never())` mean this never happened after `lb0.shutdown()`? This could pass unexpectedly if the action did happen but was before `lb0.shutdown()`. I always verify without `inOrder` just to be sure.

Reply:
Good point. Didn't check carefully when adding inOrder. Removed inOrder for this line.